ftprime/argrecorder.py

```diff
@@ -304,20 +376,61 @@ def simplify(self, samples):
+            new_right = self.edges.right[:self.last_sorted_edge:-1]
+            new_parent = self.edges.parent[:self.last_sorted_edge:-1]
+            new_child = self.edges.child[:self.last_sorted_edge:-1]
+            parent_time = self.__nodes.time[new_parent]
+            breakpoints = np.where(parent_time[1:] != parent_time[:-1])[0] + 1
+            self.__edges.reset()
+            if self.timings is not None:
+                self.timings.time_appending += timer.process_time() - before
+                before = timer.process_time()
+            start = 0
+            for end in itertools.chain(breakpoints, [-1]):
+                assert np.all(parent_time[start: end] == parent_time[start])
+                self.__edges.append_columns(left=new_left[start: end],
+                                            right=new_right[start: end],
+                                            parent=new_parent[start: end],
+                                            child=new_child[start: end])
+                msprime.sort_tables(nodes=self.__nodes,
+                                    edges=self.__edges,
+                                    sites=self.__sites,
+                                    mutations=self.__mutations,
+                                    edge_start=start)
+                start = end
+            if self.timings is not None:
+                self.time_sorting += timer.process_time() - before
+                before = timer.process_time()
+
+            # NOTE: matches fwdpy11_argexample code from
+            # https://github.com/molpopgen/fwdpy11_arg_example/pull/8
+            # Append the old sorted edges to the table.
+            self.__edges.append_columns(left=left, right=right, parent=parent,
+                                        child=child)
         if self.timings is not None:
-            start2 = timer.process_time()
-            self.timings.time_sorting += start2 - start
-        msprime.simplify_tables(samples=sample_nodes, nodes=self.nodes, 
-                                edges=self.edges, sites=self.sites, 
-                                mutations=self.mutations, 
+            before = timer.process_time()
+        msprime.simplify_tables(samples=sample_nodes, nodes=self.__nodes,
+                                edges=self.__edges, sites=self.__sites,
+                                mutations=self.__mutations,
                                 sequence_length=self.sequence_length)
         #                       migrations=self.migrations)
+        self.last_sorted_edge = self.edges.num_rows
+        # end modified block from @jeromekelleher and @molpopgen
         if self.timings is not None:
-            self.timings.time_simplifying += timer.process_time() - start2
+            self.timings.time_simplifying += timer.process_time() - before
         # update the internal state
         self.last_update_node = self.nodes.num_rows
         # update index map: sample[k] now maps to k
```

Comment:
It would be more safe to use the (undocumented) `node_map=True` argument to `simplify_tables()` that will return the node map as well (but maybe not,  see https://github.com/jeromekelleher/msprime/issues/316 ).

Reply:
er, so better not change until that issue is answered

Reply:
ok and you said the answer is 'no'. we'll find another way